Relying Party sequence diagram, examples and editorials (#91)
italia/eidas-it-wallet-docs: docs/en/relying-party-solution.rst

In this section the following flows are described:

> - **Remote Same Device Flow**, where the Web Browser and the Wallet Instance are used in the same device.

Reviewer: Why Web Browser? Couldn't it also be an app? I propose a generic name

Author: I suppose that even if embedded in the app it is always a web browser (better: user agent)

may we say just user-agent?

Reviewer: better!

Reviewer 2: In some contexts it's used the term `Reader Instance` as the _abstract_ agent that performs a single interaction with the Relying Party (that is: a Reader Instance relates to Relying Party as Wallet Intance relates to Wallet Provider).

Author: Reader Instance is defined for proximity flows while in this section we only cover remote flows

Reviewer 2: I see. It's just a proposal to use a name that does not bring any technological constraint/expectation. User-agent is fine; in same-device flow, we can have web applications (browser) or native applications.

> - **Remote Same Device Flow**, where the Web Browser and the Wallet Instance are used in the same device.

Author: ```suggestion
- **Remote Same Device Flow**, where the user-agent and the Wallet Instance are used in the same device.
```

> - **Remote Cross Device Flow**, where the Web Browser and the Wallet Instance are used in different devices.

Author: ```suggestion
- **Remote Cross Device Flow**, where the user-agent and the Wallet Instance are used in different devices.
```

In this section the following flows are described:

- **Remote Same Device Flow**, where the Web Browser and the Wallet Instance are used in the same device.
- **Remote Cross Device Flow**, where the Web Browser and the Wallet Instance are used in different devices.

In the **Same Device** and **Cross Device** Authorization Flows described in this chapter, the User interacts with a remote Relying Party.

Remote Protocol Flow
--------------------

In this scenario the Relying Party provides the URL where the signed presentation request object is available for download.

> The Relying Party MUST detect the device type of the requestor (Wallet Instance), if it is a mobile device or a workstation, and activate one the supported remote flows:
> 

Reviewer: Put this way it looks like the Wallet Instance could be a workstation as well.

```suggestion
Depending on whether the Relying Party client is on a mobile device or a workstation, it must activate one of the supported remote flows:
```
